src/utils/thread_helper.cpp

```diff
@@ -147,7 +147,7 @@ void ThreadHelper::EnergyScan(uint32_t aScanDuration, EnergyScanHandler aHandler
 exit:
     if (error != OT_ERROR_NONE)
```

Comment:
Usually, we install callbacks only when the error is `OT_ERROR_NONE` , and only then the callbacks will be invoked later. 
Why are we using a different strategy here, where callbacks seem to be called no matter it's successful or not?